std/algorithm/mutation.d

```diff
@@ -281,6 +307,14 @@ Elements can be swapped across ranges of different types:
         bringToFront(r1, r2) == 4 || assert(0);
         assert(equal(arr, [1, 2, 3, 4, 5, 6, 7]));
     }
+
+    // Bugzilla 16959
```

Comment:
In my opinion, the caller should be responsible for that since bringToFront  seems to be designed primarily for moving elements in ranges, not as a string function.